### PR TITLE
fix slice indexing of DynamicTableRegion and add test

### LINKED ISSUE #842: issue indexing dynamic table region with slice
## 1) Bug

There is currently a bug when using a `slice` to access a region of a `DynamicTableRegion`. The slice should index `self.data` which should then be used to index `self.target` (`DynamicTable`). Instead, the `slice` indexes `[0, 1, 2, ...]`.

(This was my mistake. I did something goofy and it looks like I didn't have a good understanding of slice indexing at the time. The fix is pretty straightforward.)

If you are reporting a bug please provide the following:
### Steps to Reproduce
```python
from pynwb.core import DynamicTableRegion
dtr = DynamicTableRegion('name', [0,1,1], 'desc', nwb.electrodes)
dtr[:2]
```

## Checklist

- [x] Have you ensured the feature or change was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues/) ?
- [x] Have you included a brief and descriptive title?
- [x] Have you included a clear description of the problem you are trying to solve?
- [x] Have you included a minimal code snippet that reproduces the issue you are encountering?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?

### PR DESCRIPTION
**Pull request by bendichter**
## Motivation

fix #842 

## Checklist

- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR description clearly describes problem and the solution?
- [x] Is your contribution compliant with our coding style ? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number ?

### CODE REVIEW

**Comment by bendichter** [comment]
The error is due to https://github.com/AllenInstitute/AllenSDK/issues/426.  It looks like they are aware of the problem and working to resolve it quickly.

File: tests/unit/pynwb_tests/test_core.py [hunk 1]
```diff
@@ -206,8 +206,9 @@ def test_extra_columns(self):
     def test_indexed_dynamic_table_region(self):
         table = self.with_columns_and_data()
 
-        dynamic_table_region = DynamicTableRegion('dtr', [0, 1], 'desc', table=table)
-        assert dynamic_table_region[slice(0, 1)]
+        dynamic_table_region = DynamicTableRegion('dtr', [0, 1, 1], 'desc', table=table)
+        fetch_ids = [x[1] for x in dynamic_table_region[:3]]
+        assert fetch_ids == [1, 2, 2]
```
Comment (ajtritt) [inline tests/unit/pynwb_tests/test_core.py]: Is there a reason you're not using [assertListEqual](https://docs.python.org/3.7/library/unittest.html#unittest.TestCase.assertListEqual) or [assert_array_equal](https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.testing.assert_array_equal.html)?

Comment (bendichter) [reply tests/unit/pynwb_tests/test_core.py]: no, I'm happy to use those instead. What is the advantage?

Comment (ajtritt) [reply tests/unit/pynwb_tests/test_core.py]: The TestCase assert methods give more informative error message in the event that the test fails.

Comment (bendichter) [reply tests/unit/pynwb_tests/test_core.py]: ah ok that makes sense

**Review by ajtritt — APPROVED** [review]

**Pull request merged**